.github/workflows/regression-test-with-clang.yml

```diff
@@ -0,0 +1,53 @@
+name: Regression test (AMD64 GNU/Linux Clang, unity build)
```

Comment:
As long as we are always using unity build in regression test, we can name this test `Regression test (AMD64 GNU/Linux Clang)` (drop "unity build").